lib/storage-revocation.js

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const assert = require('assert-plus');
+const bedrock = require('bedrock');
+const database = require('bedrock-mongodb');
+const {promisify} = require('util');
+const {util: {BedrockError}} = bedrock;
+
+bedrock.events.on('bedrock-mongodb.ready', async () => {
+  await promisify(database.openCollections)(['zcap-revocation']);
+
+  // FIXME: what are indexes for this collection?
+
+  await promisify(database.createIndexes)([{
+    collection: 'zcap-revocation',
+    fields: {invocationTarget: 1, id: 1},
+    options: {unique: true, background: false}
```

Comment:
There's a weird attack vector we need to figure out here with making this unique. Since control over the ID of the zcap is in the hands of the delegator, a bad delegator could find another delegator's zcap -- and redelegate a zcap for the same `invocationTarget` using the same `id` ... and then revoke it. This would prevent that other delegator from ever revoking their own zcap. Obviously that's not cool. So we need to consider things like this when deciding whether or not indexes can be unique here. If we bring in `delegator` into the compound index we can avoid that issue. The `controller` field may be playing that role here -- so perhaps `controller` should be brought in.

Reply:
I believe we decided that `delegator` and `id` for a unique compound index which is what is currently in the code.